wagtail/images/fields.py

```diff
@@ -114,11 +115,65 @@ def check_image_pixel_size(self, f):
             )
 
     def to_python(self, data):
-        f = super().to_python(data)
+        """
+        Check that the file-upload field data contains a valid image (GIF, JPG,
+        PNG, etc. -- whatever Willow supports). Overridden from ImageField to use
+        Willow instead of Pillow as the image library in order to enable SVG support.
+        """
+        f = FileField.to_python(self, data)
+        if f is None:
+            return None
+
+        # We need to get a file object for Pillow. We might have a path or we might
+        # have to read the data into memory.
+        if hasattr(data, "temporary_file_path"):
+            file = data.temporary_file_path()
+        else:
+            if hasattr(data, "read"):
+                file = BytesIO(data.read())
+            else:
+                file = BytesIO(data["content"])
+
+        try:
+            # Annotate the python representation of the FileField with the image property so subclasses can reuse it for their own validation
+            f.image = willow.Image.open(file)
+            f.content_type = image_format_name_to_content_type(f.image.format_name)
```

Comment:
To-Do: replace when https://github.com/wagtail/Willow/pull/102 is merged in

Reply:
If we can get https://github.com/wagtail/Willow/pull/102 first we can eliminate a bit of churn.